pkg/framework/didresolver/didresolver.go

```diff
@@ -73,3 +81,14 @@ func (r *DIDResolver) Resolve(did string, opts ...ResolveOpt) (*diddoc.Doc, erro
 
 	return didDoc, nil
 }
+
+// resolveDidMethod resolve did method
+func (r *DIDResolver) resolveDidMethod(method string) (DidMethod, error) {
+	for _, v := range r.didMethods {
+		match, _ := regexp.MatchString(v.id, method)
```

Comment:
The regular expression should be pre-compiled rather than compiled on demand during resolve.

Reply:
done